src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,81 @@
+import { currentTodoSlice } from '../../features/currentTodo';
+
+type Props = {
+  currentTodo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({ currentTodo }) => {
+  const { id, title, userId, completed } = currentTodo;
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    const loadUser = async () => {
+      try {
+        const data = await getUser(userId);
+
+        setUser(data);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadUser();
+  }, [userId]);
+
+  const handleModalClose = () => {
+    dispatch(currentTodoSlice.actions.clearTodo());
+  };
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{id}
+            </div>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={handleModalClose}
+            />
+          </header>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {title}
+            </p>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
-
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+            <p className="block" data-cy="modal-user">
+              {!completed ? (
+                <strong className="has-text-danger">Planned</strong>
+              ) : (
+                <strong className="has-text-success">Done</strong>
+              )}
+              {' by '}
+              <a href="mailto:Sincere@april.biz">{user?.name}</a>
```

Comment:
The email address in the anchor tag is hardcoded as 'Sincere@april.biz'. Consider replacing it with the user's actual email from the fetched data, like this: `<a href={user ? `mailto:${user.email}` : '#'}>{user?.name}</a>`.